lib/rubocop/cop/gemspec/consistent_assignment.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Gemspec
+      # Use consistent style for Gemspec attributes assignment.
+      #
+      # @example
+      #
+      #   # bad
+      #   Gem::Specification.new do |spec|
+      #     spec.metadata = { 'key' => 'value' }
+      #     spec.metadata['another-key'] = 'another-value'
+      #   end
+      #
+      #   # good
+      #   Gem::Specification.new do |spec|
+      #     spec.metadata = { 'key' => 'value', 'another-key' => 'another-value' }
+      #   end
```

Comment:
I don't understand why this style of assignment using a hash literal is preferred. Specifically, the gemspec generated by running `bundle gem` uses `spec.metadata[key] = value` style.

```console
$ bundle gem foo
$ cd foo
$ cat foo.gemspec
(snip)
  spec.metadata["allowed_push_host"] = "TODO: Set to your gem server 'https://example.com'"

  spec.metadata["homepage_uri"] = spec.homepage
  spec.metadata["source_code_uri"] = "TODO: Put your gem's public repo URL here."
  spec.metadata["changelog_uri"] = "TODO: Put your gem's CHANGELOG.md URL here."
```

Therefore, I don't see a clear reason why a different hash literal style would be preferable as a coding style, especially when it differs from the default used in gemspec.

Reply:
This was just a bad (incomplete) example. The kind of code is provided is fine (the cop does not detect any offenses) unless there's an extra `spec.metadata = { ... }` assignment somewhere

Reply:
Hm, either way, it might be better to get it accepted in the style guide first.
https://github.com/rubocop/ruby-style-guide?tab=readme-ov-file#gemfile-and-gemspec